### DISCUSSION TITLE
Failed to retrieve rows from table

### DISCUSSION THREAD

**Discussion by rrcanonigo** [supabase/supabase#28131] [Questions] [answered]
![Screenshot 2024-07-23 091813](https://github.com/user-attachments/assets/171e68e3-2d65-4a86-9ff7-af7366f9b7bc)

Hi,

anon and authenticated user roles get this error but postgres does not. I have enabled RLS to allow anon and authenticated user to select query but still getting this error.

**Comment by GaryAustin1** [comment, marked as answer]
The error is implying your public schema has been modified to not allow anon/authenticated/service_role to access it.
This is very common with Prisma.
See troubleshooting here:  https://supabase.com/partners/integrations/prisma
Otherwise more info on what you are doing when you get the error and your setup is needed.

**Reply by rrcanonigo** [reply]
@GaryAustin1 checked the documentation and apparently, the grants worked just fine! thank you!

**Reply by GaryAustin1** [reply]
Can you confirm you were using Prisma or something needed grants fixed?

**Reply by rrcanonigo** [reply]
@GaryAustin1 

Wasn't using Prisma. The tables just needed grants. lol

**Reply by GaryAustin1** [reply]
That it is odd, as that is default setup.  
Something to watch out for in future if you did not purposely change them.

**Comment by liemdo** [comment]
I encountered this issue after restoring the database from another service to Supabase. The tables are not exposed to GraphQL. Granting the correct permissions to the `public` schema resolves the problem.

```bash
grant usage on schema public to anon, authenticated, service_role;
grant all privileges on all tables in schema public to anon, authenticated, service_role;
grant all privileges on all routines in schema public to anon, authenticated, service_role;
grant all privileges on all sequences in schema public to anon, authenticated, service_role;
alter default privileges for role postgres in schema public grant all on tables to anon, authenticated, service_role;
alter default privileges for role postgres in schema public grant all on routines to anon, authenticated, service_role;
alter default privileges for role postgres in schema public grant all on sequences to anon, authenticated, service_role;
```

The grant referenced above can be found here: https://supabase.com/docs/guides/graphql#add-a-schema-to-the-api.
